### DISCUSSION TITLE
Install script fails with `brew: No such file or directory`

### DISCUSSION THREAD

**Discussion by infotexture** [Homebrew/discussions#4174] [Getting started] [answered]
### Output of `brew config`

```shell
brew: command not found
```

### Output of `brew doctor`

```shell
brew: command not found
```

### Description of issue

Up until a few days ago, I had a healthy Homebrew installation that was humming along fine until `brew doctor` told me to check out `master` as described in #4120.

I found this puzzling, but followed the instructions, and now the `brew` command itself is no longer found, although the ARM-native installation location `/opt/homebrew/bin` is first in my PATH.

I went looking for the executable there, and it is indeed missing, so I re-ran the installation script per https://brew.sh:

```bash
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

— but oddly enough, the install script fails and complains that it cannot find the `brew` binary it's meant to install:

```bash
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R username:admin /opt/homebrew
==> Downloading and installing Homebrew...
HEAD is now at 39e158fb9 Merge pull request #14501 from p-linnane/cookbook-update
/opt/homebrew/Library/Homebrew/brew.sh: line 155: /opt/homebrew/bin/brew: No such file or directory
Failed during: /opt/homebrew/bin/brew update --force --quiet
```

I assume this is somehow related to the recent architectural changes from https://github.com/Homebrew/brew/pull/14359, but I can't figure out how to recover if I can't re-install.

I still have my local [Homebrew/homebrew-core](https://github.com/Homebrew/homebrew-core) tap at `/opt/homebrew/Library/Taps/homebrew/homebrew-core`, and that pulls the latest changes from the upstream `master` branch, but I still can't re-install `brew`.

— Any idea what's going on here, and how to recover? 🤔

**Comment by infotexture** [comment]
Tried setting the beta `HOMEBREW_INSTALL_FROM_API` environment variable per 
https://docs.brew.sh/Installation#skip-tap-cloning-beta, but that doesn't seem to change anything.

**Reply by SMillerDev** [reply]
How about `HOMEBREW_NO_INSTALL_FROM_API` ?

**Reply by infotexture** [reply]
Yeah, looks like the `HOMEBREW_INSTALL_FROM_API` option has since been removed from the docs in favor of `HOMEBREW_NO_INSTALL_FROM_API`.

Tried setting that per the [Default Tap Cloning](https://docs.brew.sh/Installation#default-tap-cloning) section, but neither `0` nor `1` changes anything either.

**Reply by SMillerDev** [reply]
What's currently in that directory?

**Reply by gromgit** [reply]
And what's the output of
```
env | grep HOMEBREW_
```
?

**Reply by infotexture** [reply]
> What's currently in that directory?

@SMillerDev Which directory do you mean? 🤔 

If it's this, it's just whatever earlier Homebrew versions cloned there:

> I still have my local [Homebrew/homebrew-core](https://github.com/Homebrew/homebrew-core) tap at `/opt/homebrew/Library/Taps/homebrew/homebrew-core`

If I navigate there with a Git client, I can fetch & pull as usual, so the repo seems intact.

**Reply by infotexture** [reply]
> And what's the output of `env | grep HOMEBREW_` ?

```
bash-5.2$ env | grep HOMEBREW_
HOMEBREW_NO_INSTALL_FROM_API=1
```

Tried setting that to `0` instead, but the results are the same.

**Comment by cho-m** [comment, marked as answer]
Sounds like your git repo isn't clean.
```
/opt/homebrew/Library/Homebrew/brew.sh: line 155: /opt/homebrew/bin/brew: No such file or directory
```

`bin/brew` should be maintained by git (https://github.com/Homebrew/brew/blob/master/bin/brew). If it is missing, then there is something wrong with the state of the git repo.

---

Try going to the repo directory and checking status, e.g.
```console
❯ cd /opt/homebrew

❯ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

If output shows something other than above, then it will probably need to be manually fixed.

**Reply by infotexture** [reply]
> Sounds like your git repo isn't clean.

💯 That was indeed the problem. 

I had been looking at `/opt/homebrew/Library/Taps/homebrew/homebrew-core`, but didn't think to verify `/opt/homebrew`. 

When I looked there, I found that it was checked out to an outdated `stable` branch state, and contained a stash entry `WIP on stable: e4c18be30 …` that had apparently been generated a few days ago when I was prompted to stash local changes.

Resetting to `master` fixed everything, so I unset the new environment variables.

Thanks for your help! 🙏

**Reply by JideijeEmeka** [reply]
See what I get as response
How did you reset to master that solved the issue?...Please help
<img width="514" alt="Screenshot 2024-02-03 at 13 19 04" src="https://github.com/Homebrew/discussions/assets/75017328/d8bbd017-6b8d-471d-90ec-5263364c3311">

**Reply by gromgit** [reply]
It would be highly unusual for the whole of `/usr/local` to be under Git management.

You want to `cd /usr/local/Homebrew` instead, assuming you're on an Intel Mac.
